nifi-nar-bundles/nifi-questdb-bundle/nifi-questdb-status-history/src/main/java/org/apache/nifi/controller/status/history/questdb/QuestDbStatusHistoryStorage.java

```diff
@@ -0,0 +1,181 @@
+        return getResult(query, new GarbageCollectionResultProcessor(), Collections.emptyList());
+    }
+
+    @Override
+    public List<StatusSnapshot> getNodeStatusSnapshots(final Date start, final Date end) {
+        final String storageStatusQuery = String.format(STORAGE_STATUS_QUERY, getStartTime(start), getEndTime(end));
+        final Map<Long, Map<StandardMetricDescriptor<NodeStatus>, Long>> statusMetricsByTime
+            = getResult(storageStatusQuery, EmbeddedQuestDbStatusHistoryRepositoryDefinitions.getStorageStatusResultProcessor(), new HashMap<>());
+        final String nodeStatusQuery = String.format(NODE_STATUS_QUERY, getStartTime(start), getEndTime(end));
+        return getSnapshot(nodeStatusQuery, EmbeddedQuestDbStatusHistoryRepositoryDefinitions.getNodeStatusResultProcessor(statusMetricsByTime));
+    }
+
+    @Override
+    public void storeNodeStatuses(final Collection<CapturedStatus<NodeStatus>> statuses) {
+        store(TABLE_NAME_NODE_STATUS, EmbeddedQuestDbStatusHistoryRepositoryDefinitions.getNodeStatusDataSource(statuses));
+        store(TABLE_NAME_STORAGE_STATUS, StorateStatusDataSource.getInstance(statuses));
+    }
+
+    @Override
+    public void storeGarbageCollectionStatuses(final Collection<CapturedStatus<GarbageCollectionStatus>> statuses) {
+        store(TABLE_NAME_GARBAGE_COLLECTION_STATUS, EmbeddedQuestDbStatusHistoryRepositoryDefinitions.getGarbageCollectionStatusDataSource(statuses));
+    }
+
+    @Override
+    public void storeProcessGroupStatuses(final Collection<CapturedStatus<ProcessGroupStatus>> statuses) {
+        store(TABLE_NAME_PROCESS_GROUP_STATUS, EmbeddedQuestDbStatusHistoryRepositoryDefinitions.getProcessGroupStatusDataSource(statuses));
+    }
+
+    @Override
+    public void storeConnectionStatuses(final Collection<CapturedStatus<ConnectionStatus>> statuses) {
+        store(TABLE_NAME_CONNECTION_STATUS, EmbeddedQuestDbStatusHistoryRepositoryDefinitions.getConnectionStatusDataSource(statuses));
+    }
+
+    @Override
+    public void storeRemoteProcessorGroupStatuses(final Collection<CapturedStatus<RemoteProcessGroupStatus>> statuses) {
+        store(TABLE_NAME_REMOTE_PROCESS_GROUP_STATUS, EmbeddedQuestDbStatusHistoryRepositoryDefinitions.getRemoteProcessGroupStatusDataSource(statuses));
+    }
+
+    @Override
+    public void storeProcessorStatuses(final Collection<CapturedStatus<ProcessorStatus>> statuses) {
+        store(TABLE_NAME_PROCESSOR_STATUS, EmbeddedQuestDbStatusHistoryRepositoryDefinitions.getProcessorStatusDataSource(statuses));
+        store(TABLE_NAME_COMPONENT_COUNTER, EmbeddedQuestDbStatusHistoryRepositoryDefinitions.getCounterStatisticsDataSource(statuses));
+    }
+
+    private <T> void store(final String tableName, final InsertRowDataSource source) {
+        try {
+            client.insert(tableName, source);
+        } catch (final DatabaseException e) {
+            LOGGER.error("Error during storing snapshots to " + tableName, e);
```

Comment:
Placeholders should be used instead of string concatenation for logs.
```suggestion
            LOGGER.error("Error during storing snapshots to table [{}]", tableName, e);
```